sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -127,8 +127,10 @@ object TypeCoercion {
     case (DateType, TimestampType) => Some(StringType)
     case (StringType, NullType) => Some(StringType)
     case (NullType, StringType) => Some(StringType)
+    case (StringType, IntegerType) => Some(DoubleType)
```

Comment:
How abount other types (e.g., LongType)? Is the similar issue still exists?
I think the logic in pr-15880 (https://github.com/apache/spark/pull/15880) is reasonable.